transaction-coordinator/src/main/java/org/apache/kafka/coordinator/transaction/TransactionLogConfig.java

```diff
@@ -16,11 +16,43 @@
  */
 package org.apache.kafka.coordinator.transaction;
 
+import org.apache.kafka.common.config.TopicConfig;
+import org.apache.kafka.server.config.ServerTopicConfigSynonyms;
+import org.apache.kafka.storage.internals.log.ProducerStateManagerConfig;
+
 public class TransactionLogConfig {
```

Comment:
Adding "s" to the name make sense but seems like we don't have one pattern in Kafka for example there is SaslConfigs and TopicConfig. I'll rename it but we properly need to update https://issues.apache.org/jira/browse/KAFKA-14524 to add some guidance for future refactors

Reply:
> add private constructor

Why do we need this? we usually don't add it if the config class just to group the props, docs and default values.

Reply:
> Why do we need this? we usually don't add it if the config class just to group the props, docs and default values.

that is used to avoid creating instance, and that is a guideline of `Effective Java book`

anyway, that is just a code style suggestion, so it is fine to ignore it :)